### PR TITLE
copr: call post_apply hook

### PR DESCRIPTION
**Pull request by BusyJay**
We need to ensure `post_apply` is called.

### CODE REVIEW

File: src/raftstore/store/worker/apply.rs [hunk 1]
```diff
@@ -321,6 +348,7 @@ impl ApplyDelegate {
         if committed_entries.is_empty() {
             return vec![];
         }
+        apply_ctx.prepare_for(self);
```
Comment (siddontang) [inline src/raftstore/store/worker/apply.rs]: why prepare here?

Comment (BusyJay) [reply src/raftstore/store/worker/apply.rs]: Add a new `ApplyCallback` struct for current region.

File: src/raftstore/store/worker/apply.rs [hunk 2]
```diff
@@ -167,10 +167,33 @@ pub enum ExecResult {
     DeleteRange { ranges: Vec<Range> },
 }
 
+struct ApplyCallBack {
```
Comment (siddontang) [inline src/raftstore/store/worker/apply.rs]: Callback may be better?

Comment (BusyJay) [reply src/raftstore/store/worker/apply.rs]: There is a `Callback` already.

Comment (siddontang) [reply src/raftstore/store/worker/apply.rs]: I mean ApplyCallback, you use the name Callback in other places.

**Pull request merged**